### DISCUSSION TITLE
Is RedisStorageProvider still in BETA?

### DISCUSSION THREAD

**Discussion by mohan-venkatakrishnan** [jobrunr/jobrunr#735] [General]
As per the official documentation, https://www.jobrunr.io/en/documentation/installation/storage/ RedisStorageProvider is mentioned to be in BETA and not recommended for production use. Redis support in jobrunr was launched way back in April, 2020 and would like to know if this is still in BETA. 
If yes, can we please know what are the limitations that we know of, so that we can take a call on implementing this for production usage.

Thanks.

**Comment by vishwa6990** [comment]
Would appreciate if we could get any response here. As we are planning to use redis as storage provider for our production usage, we need to be sure of any known issues with it.
Found a [comment](https://github.com/jobrunr/jobrunr/issues/429#issuecomment-1110173632) with recurring jobs & redis issue highlighting that it was the primary reason of marking it as beta but now it is closed as fixed. 
Is there any known issues with respect to redis usage.

We are currently in the evaluation phase and if Jobrunr + Redis seem to solve our use case & traffic requirements, then we are looking to explore on the Pro version of it.

**Comment by rdehuyss** [comment]
Well, personally I'm not happy with the Redis `StorageProvider` implementation. It's difficult to maintain and there are not a lot of users using Redis. See also my [blog post](https://www.jobrunr.io/en/blog/2023-03-16-10-billion-jobs/) about JobRunr having processed 10 billion jobs.

I'll copy the relevant paragraph here:
> Clear is that Postgres is the king 👑 of databases with over 1,323 active installs, followed by MongoDB with 416 installs, MariaDB with 401 installs, MySQL with 340 installs, H2 with 277 installs, Oracle with 144 installs, Microsoft SQL Server with 137 installs, Redis using Lettuce with 62 installs, Redis using Jedis with 54 installs, and a few users with custom database implementations.

That being said, I know of working JobRunr installations using Redis.

Ideally, the StorageProvider would be reimplemented using [Redis Stack](https://redis.io/docs/stack/) which would make for a better `StorageProvider` implementation but as usage of Redis is low, this is low priority for me.

I hope this helps.

**Reply by rdehuyss** [reply]
Question: is anybody using Redis Stack already and is it an option for you?

**Comment by rdehuyss** [comment]
Question @arielazzi - are you using RedisStack already?

**Reply by arielazzi** [reply]
@rdehuyss No, currently we're just using Redis OSS for some cache implementations. Could you please describe why it is still in beta? Could it be because it doesn't support Redis stack? Or is there some other issue that we should be concerned about?

**Reply by rdehuyss** [reply]
All the reasons are mentioned above... I don't know what you expect me to add.
